libecole/include/ecole/observation/milpbipartite.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <optional>
+
+#include <xtensor/xtensor.hpp>
+
+#include "ecole/observation/abstract.hpp"
+#include "ecole/utility/sparse-matrix.hpp"
+
+namespace ecole::observation {
+
+class MilpBipartiteObs {
+public:
+	using value_type = double;
+
+	static inline std::size_t constexpr n_variable_features = 9;
+	enum struct VariableFeatures : std::size_t {
+		objective = 0,
+		is_type_binary,            // One hot encoded
+		is_type_integer,           // One hot encoded
+		is_type_implicit_integer,  // One hot encoded
+		is_type_continuous,        // One hot encoded
+		has_lower_bound,
+		has_upper_bound,
+        lower_bound,
+        upper_bound,
+	};
+	static inline std::size_t constexpr n_constraint_features = 1;
+    enum struct ConstraintFeatures : std::size_t {
+		bias = 0,
+	};
+
+	xt::xtensor<value_type, 2> variable_features;
+	xt::xtensor<value_type, 2> constraint_features;
+	utility::coo_matrix<value_type> edge_features;
+};
+
+class MilpBipartite : public ObservationFunction<std::optional<MilpBipartiteObs>> {
+public:
+	MilpBipartite(bool use_normalization = false) : normalize{use_normalization}  {}
+
+	std::optional<MilpBipartiteObs> extract(scip::Model& model, bool done) override;
+    
+private:
+    bool normalize = true;
```

Comment:
It is weird that `normalize` has default value `true` here, and `false` in the constructor. It is weird also that it's default value is defined at two different places. I would either not set its value here, or not having a default value in the constructor, and have an additional constructor with no arguments.

Reply:
I removed the default value there.